File: 3. Baekjoon/1924. 2007년.py
import sys

input = sys.stdin.readline


if __name__ == "__main__":
    day = 0
    month = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    week = ["SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"]

    # 월, 일
    x, y = map(int, input().split())

    # x월까지 순회해서 day 카운트
    for i in range(x - 1):
        day += month[i]

    # mod 연산으로 요일 탐색
    day = (day + y) % 7

    print(week[day])

File: 3. Baekjoon/~25.04.01/1924. 2007년.cpp
#include <iostream>
#include <vector>

using namespace std;

typedef vector<int> MonthDays;
typedef vector<string> WeekDays;

int main() {
    int day = 0;
    MonthDays month = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    WeekDays week = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};

    int x, y;
    cin >> x >> y;

    // x월까지 순회해서 day 카운트
    for (int i = 0; i < x - 1; i++) {
        day += month[i];
    }

    // mod 연산으로 요일 탐색
    day = (day + y) % 7;

    cout << week[day] << endl;

    return 0;
}
